File: app/src/main/java/com/code_design_camp/client/friday/HeadDisplayClient/fragments/net/ConnectionFragment.java
package com.code_design_camp.client.friday.HeadDisplayClient.fragments.net;

import android.util.Log;

import androidx.fragment.app.Fragment;

/**
 * Extendable Fragment class to automatically implement {@link OnConnectionStateChangedListener}
 * This class is necessary to properly access unknown subclasses
 * @see OnConnectionStateChangedListener
 * @see com.code_design_camp.client.friday.HeadDisplayClient.fragments.store.ErrorFragment
 */
public class ConnectionFragment extends Fragment implements OnConnectionStateChangedListener {

    @Override
    public void onConnected() {

    }

    @Override
    public void onError(Exception e) {
        Log.e("ConnectionFragment", e.getMessage());
    }
}

File: appModules/core/src/main/java/com/friday/ar/core/net/ConnectionFragment.kt
package com.friday.ar.core.net

import android.util.Log

import androidx.fragment.app.Fragment
import extensioneer.isNull
import extensioneer.notNull

/**
 * Extendable Fragment class to automatically implement [OnConnectionStateChangedListener]
 * This class is necessary to properly access unknown subclasses
 *
 * @see OnConnectionStateChangedListener
 */
open class ConnectionFragment : Fragment(), OnConnectionStateChangedListener {

    override fun onConnected() {

    }

    override fun onError(e: Exception) {
        e.message.notNull {
            Log.d("ConnectionFragment", "ErrorCallback error:$this in fragment ${this@ConnectionFragment}")
        }.isNull {
            Log.d("ConnectionFragment", "error in ConnectionFragment: ${this@ConnectionFragment}")
        }
    }
}

